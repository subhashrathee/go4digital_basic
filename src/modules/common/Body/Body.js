import React from 'react'
import PropTypes from 'prop-types'
import HtmlToReact from 'html-to-react'

var HtmlToReactParser = HtmlToReact.Parser;
var htmlToReactParser = new HtmlToReactParser();

/**
* Body component which takes in a content prop and parses it to render the content.
*
* Props: 
* @String content A content string which is rendered in the body of the page
* @String heading A content string which is rendered above the body of the page
*
* State vars: None
*/
class Body extends React.Component {
  render() {

  var reactElement = htmlToReactParser.parse(this.props.content.toString());

    return (
      <div className="main">
        <h1>
        {this.props.heading} 
        </h1>

        {reactElement}
      </div>
    )
  }
}

Body.defaultProps = {
  content: '<p>Default body content</p>',
  heading: 'go4digital website and marketing agency'
}

Body.propTypes = {
  content: PropTypes.string,
  heading: PropTypes.string
}

export default Body;
