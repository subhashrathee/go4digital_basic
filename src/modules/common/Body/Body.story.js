import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { BrowserRouter } from 'react-router-dom'
import Body from './Body'

storiesOf('Body', module)
  .add('Default Body', () => (
      <Body />
  ))
  .add('Body with test content', () => (
      <Body heading="A test heading" content="<p>Some test content</p>" />
  ))
;
