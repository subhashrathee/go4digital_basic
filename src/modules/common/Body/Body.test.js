import React from 'react';
import Body from './Body';
import renderer from 'react-test-renderer';
import { BrowserRouter, Link } from 'react-router-dom'

test('Body component', () => {
  const component = renderer.create(
    <Body />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
