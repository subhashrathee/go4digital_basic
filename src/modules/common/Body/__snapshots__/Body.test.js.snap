// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body component 1`] = `
<div
  className="main"
>
  <h1>
    go4digital website and marketing agency
  </h1>
  <p>
    Default body content
  </p>
</div>
`;
